src/Drupal/Driver/Cores/Drupal8.php

```diff
@@ -349,13 +350,28 @@ public function getExtensionPathList() {
   }
 
   /**
+   * Expands specified base fields on the given entity object to the expected structure.
+   *
+   * @param string $entity_type
+   *   The entity type for which to return the field types.
+   * @param \stdClass $entity
+   *   Entity object.
+   * @param array $base_fields
+   *   Base fields to be expanded in addition to user defined fields.
+   */
+  public function expandEntityBaseFields($entity_type, \stdClass $entity, $base_fields) {
```

Comment:
This new method is public. `expandEntityFields()` cannot be used for custom entities because it is protected.